src/commands/create.js

```diff
@@ -0,0 +1,265 @@
+		}
+
+		if (exists && argv.force) {
+			if (! await prompt({
+				type: 'confirm',
+				name: 'enableForce',
+				message: `You are using '--force'. Do you wish to continue and overwrite ${target}?`,
+				default: false
+			})) {
+				return error('Refusing to overwrite current directory!', 1);
+			}
+		}
+
+		let repo = argv.template;
+		if (!repo.includes('/')) {
+			repo = `${ORG}/${repo}`;
+			info(`Fetching ${argv.template} template from ${repo}...`);
+		}
+
+		// Use `--name` value or `dest` dir's name
+		argv.name = argv.name || argv.dest;
+
+		let { errors } = isValidName(argv.name);
+		if (errors) {
+			errors.unshift(`Invalid package name: ${argv.name}`);
+			return error(errors.map(capitalize).join('\n  ~ '), 1);
+		}
+
+		// Attempt to fetch the `template`
+		let archive = await gittar.fetch(repo).catch(err => {
+			err = err || { message: 'An error occured while fetching template.' };
+			return error(err.code === 404 ? `Could not find repository: ${repo}` : err.message, 1);
+		});
+
+		let spinner = ora({
+			text: 'Creating project',
+			color: 'magenta'
+		}).start();
+
+		// Extract files from `archive` to `target`
+		// TODO: read & respond to meta/hooks
+		let keeps=[];
+		await gittar.extract(archive, target, {
+			strip: 2,
+			filter(path, obj) {
+				if (path.includes('/template/')) {
+					obj.on('end', () => {
+						if (obj.type === 'File' && !isMedia(obj.path)) {
+							keeps.push(obj.absolute);
+						}
+					});
+					return true;
+				}
+			}
+		});
+
+		if (keeps.length) {
+			let dict = new Map();
+
+			// TODO: concat author-driven patterns
+			//globally replace instances of {{name}} with value of name arg
+			['name'].forEach(str => {
+				// if value is defined
+				if (argv[str] !== void 0) {
+					dict.set(new RegExp(`{{\\s?${str}\\s}}`, 'g'), argv[str]);
+				}
+			});
+
+			// Update each file's contents
+			let buf, entry, enc='utf8';
+			for (entry of keeps) {
+				buf = fs.readFileSync(entry, enc);
+				dict.forEach((v, k) => {
+					buf = buf.replace(k, v);
+				});
+				try {
+					fs.writeFileSync(entry, buf, enc);
+				}
+				catch (err) {
+					return error(`Couldn't write ${entry}: ${err}`);
+				}
+
+			}
+		}
+		else {
+			return error(`No \`template\` directory found within ${ repo }!`, 1);
+		}
+
+		spinner.succeed().start('Parsing `package.json` file');
+
+		// Validate user's `package.json` file
+		let pkgData, pkgFile=resolve(target, 'package.json');
+
+		if (pkgFile) {
+			try {
+				pkgData = JSON.parse(fs.readFileSync(pkgFile));
+			}
+			catch (err) {
+				return error(`Could not reach package.json file: ${err}`);
+			}
+			// Write default "scripts" if none found
+			pkgData.scripts = pkgData.scripts || (await addScripts(pkgData, target, isYarn));
+		}
+		else {
+			warn('Could not locate `package.json` file!');
+		}
+
+		// Update `package.json` key
+		if (pkgData) {
+			spinner.succeed().start('Updating `name` within `package.json` file');
```

Comment:
I would like to see this include the generated name so the user is sure they generated it correctly, e.g.

```
'Updating `name` to `my-package` within `package.json` file'
```

Reply:
- [ ] add actual name to replacement status message